staging/src/k8s.io/apiserver/pkg/util/peerproxy/peerproxy.go

```diff
@@ -17,51 +17,128 @@ limitations under the License.
 package peerproxy
 
 import (
+	"context"
+	"fmt"
 	"net/http"
-	"sync"
+	"strings"
+	"time"
 
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apiserver/pkg/reconcilers"
-	"k8s.io/apiserver/pkg/storageversion"
-	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/transport"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	apidiscoveryv2 "k8s.io/api/apidiscovery/v2"
+	v1 "k8s.io/api/coordination/v1"
+	schema "k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	coordinationv1informers "k8s.io/client-go/informers/coordination/v1"
 )
 
-// Interface defines how the Unknown Version Proxy filter interacts with the underlying system.
+const localDiscoveryRefreshInterval = 30 * time.Minute
```

Comment:
Document why we need to refresh local discovery

Reply:
Done